src/test/java/io/stargate/sgv2/jsonapi/api/v1/FindOneAndReplaceIntegrationTest.java

```diff
@@ -336,6 +336,61 @@ public void withUpsert() {
           .body("data.documents[0]", jsonEquals(expected));
     }
 
+    @Test
+    public void withUpsertNewId() {
+      final String newId = "new-id-1234";
+      String json =
+          """
+                {
+                  "findOneAndReplace": {
+                    "filter" : {},
+                    "replacement" : {
+                      "username": "aaronm",
+                      "_id": "%s"
+                    },
+                    "options" : {
+                      "returnDocument": "after",
+                      "upsert": true
+                    }
+                  }
+                }
+                """
+              .formatted(newId);
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(json)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, collectionName)
+          .then()
+          .statusCode(200)
+          .body("errors", is(nullValue()))
+          .body("status.matchedCount", is(0))
+          .body("status.modifiedCount", is(0))
+          .body("data.document._id", is(newId))
+          // Should we return id of new document as upsertedId?
+          .body("status.upsertedId", is(newId));
```

Comment:
This is something I am not 100% sure about; should there instead be `insertedId`, or is it fine for this to cover both replacement of existing and insert of now document?

Reply:
I think `upsertedId` is fine